server.py

```diff
@@ -35,40 +35,46 @@
 ######################################################################
 @app.route('/')
 def index():
-    pets_url = request.base_url + "pets"
-    return jsonify(name='Pet Demo REST API Service',
-                   version='1.0',
-                   url=pets_url
+    recommendations_url = request.base_url + "recommendations"
+    return jsonify(name='Recommendations REST API Service',
+                   version='0.1',
+                   url=recommendations_url
                    ), HTTP_200_OK
 
 ######################################################################
 # LIST ALL PETS
 ######################################################################
-@app.route('/pets', methods=['GET'])
-def list_pets():
-    results = pets.values()
-    kind = request.args.get('kind')
-    if kind:
-        results = []
-        for key, value in pets.iteritems():
-            if value['kind'] == kind:
-                results.append(pets[key])
+@app.route('/recommendations', methods=['GET'])
+def list_recommendations():
+    """
+    Basically prints out all the recommendations.
+    """
 
-    return reply(results, HTTP_200_OK)
+    result_dict = {}
+    for prod_id in data:
+        recomm_list = []
+        for recomm in data[prod_id]['recommendations']:
+            recomm_list.append(recomm['name'])
+        result_dict[data[prod_id]['name']] = recomm_list
+    return reply(result_dict, HTTP_200_OK)
 
 ######################################################################
 # RETRIEVE A PET
 ######################################################################
-@app.route('/pets/<int:id>', methods=['GET'])
-def get_pets(id):
-    if pets.has_key(id):
-        message = pets[id]
+@app.route('/recommendations/<id>', methods=['GET'])
+def get_recommendations(id):
+    '''
+    Given a Product ID, Output a list of it's reccommendations.
```

Comment:
Yeah I agree it should be a JSON format with a placeholder ['data'] as the key
{
"data": [
{
"id": "8d7ebb3e",
"name": "Round-Touch",
"priority": 0
},
{
"id": "a4416148",
"name": "Freshnix",
"priority": 1
}